mixin/dashboards/ebs.libsonnet

```diff
@@ -0,0 +1,253 @@
+local common = import 'common.libsonnet';
+local grafana = import 'grafonnet-7.0/grafana.libsonnet';
+
+local allLabels = 'scrape_job=~"$job", region=~"$region", dimension_VolumeId=~"$volume"';
+
+grafana.dashboard.new(
+  title='AWS EBS',
+  description='Visualize Amazon EBS metrics',
+  tags=['Amazon', 'AWS', 'CloudWatch', 'EBS'],
+  graphTooltip=common.tooltipSharedCrosshair,
+)
+.addTemplate(
+  grafana.template.datasource.new(
+    name='datasource',
+    query='prometheus',
+    label='Data Source',
+  )
+)
+.addTemplate(
+  grafana.template.query.new(
+    name='job',
+    label='job',
+    datasource='$datasource',
+    query='label_values(aws_ebs_info, scrape_job)',
+    refresh=common.refreshOnPageLoad,
+    includeAll=true,
+    multi=true,
+    sort=common.sortAlphabeticalAsc,
+    allValue='.+',
+  )
+)
+.addTemplate(
+  grafana.template.query.new(
+    name='region',
+    label='Region',
+    datasource='$datasource',
+    query='label_values(aws_ebs_volume_idle_time_average, region)',
+    refresh=common.refreshOnTimeRangeChange,
+    includeAll=true,
+    multi=true,
+    sort=common.sortAlphabeticalAsc,
+  )
+)
+.addTemplate(
+  grafana.template.query.new(
+    name='volume',
+    label='Volume',
+    datasource='$datasource',
+    query='label_values(aws_ebs_volume_idle_time_average{scrape_job=~"$job", region=~"$region"}, dimension_VolumeId)',
+    refresh=common.refreshOnTimeRangeChange,
+    includeAll=true,
+    multi=true,
+    sort=common.sortAlphabeticalAsc,
+  )
+)
+.addPanels(
+  [
+    grafana.panel.text.new(
+      title='Info',
+      content=|||
+        Showing metrics only for AWS resources that have tags assigned to them. For more information, see [Amazon CloudWatch Metrics for Amazon EBS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using_cloudwatch_ebs.html).
```

Comment:
Hmm this copy only makes sense for `discoveryJobs` right? I'm wondering how we could make these dashboards more generic yet customisable. Maybe we should find a way to make better use of `config.libsonnet`.

Reply:
True but I think this is a good first start.